sources/packages/backend/apps/api/src/route-controllers/overaward/overaward.institution.controller.ts

```diff
@@ -0,0 +1,54 @@
+import { Controller, Param, ParseIntPipe, Get } from "@nestjs/common";
+import { ApiNotFoundResponse, ApiTags } from "@nestjs/swagger";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { AllowAuthorizedParty } from "../../auth/decorators";
+import { ClientTypeBaseRoute } from "../../types";
+import BaseController from "../BaseController";
+import {
+  OverawardBalanceAPIOutDTO,
+  OverawardDetailsAPIOutDTO,
+} from "./models/overaward.dto";
+import { OverawardControllerService } from "..";
+
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@Controller("overaward")
+@ApiTags(`${ClientTypeBaseRoute.Institution}-overaward`)
+export class OverawardInstitutionController extends BaseController {
+  constructor(
+    private readonly overawardControllerService: OverawardControllerService,
+  ) {
+    super();
+  }
+
+  /**
+   * Get the overaward balance of a student.
+   * @param studentId student.
+   * @returns overaward balance for student.
+   */
+  @ApiNotFoundResponse({
+    description: "Student not found.",
+  })
+  @Get("student/:studentId/balance")
+  async getOverawardBalance(
+    @Param("studentId", ParseIntPipe) studentId: number,
+  ): Promise<OverawardBalanceAPIOutDTO> {
+    return this.overawardControllerService.getOverawardBalance(studentId);
+  }
+
+  /**
+   * Get all overawards which belong to a student.
+   * @param studentId student.
+   * @returns overaward details of a student.
+   */
+  @ApiNotFoundResponse({
+    description: "Student not found.",
+  })
+  @Get("student/:studentId")
+  async getOverawardsByStudent(
+    @Param("studentId", ParseIntPipe) studentId: number,
+  ): Promise<OverawardDetailsAPIOutDTO[]> {
+    return this.overawardControllerService.getOverawardsByStudent(studentId, {
```

Comment:
I believe that the idea was to show the institution a view closer to what the students can see. I am not sure if the `audit` would be enabled for the public institutions as per the below AC.
"_Essentially display the same as the Student View of this tab without the top content._"

Reply:
Please remove the "Added By" column for the institution view.